kong/api/routes/consumers.lua

```diff
@@ -32,5 +32,48 @@ return {
     DELETE = function(self, dao_factory)
       crud.delete(self.consumer, dao_factory.consumers)
     end
-  }
+  },
+
+  ["/consumers/:username_or_id/plugins/"] = {
+    before = function(self, dao_factory, helpers)
+      self.params.username_or_id = ngx.unescape_uri(self.params.username_or_id)
+      crud.find_consumer_by_username_or_id(self, dao_factory, helpers)
+      self.params.consumer_id = self.consumer.id
+    end,
+
+    GET = function(self, dao_factory)
+      crud.paginated_set(self, dao_factory.plugins)
+    end,
+
+    POST = function(self, dao_factory)
+      crud.post(self.params, dao_factory.plugins)
+    end,
+
+    PUT = function(self, dao_factory)
+      crud.put(self.params, dao_factory.plugins)
+    end
+  },
+
+  ["/consumers/:username_or_id/plugins/:id"] = {
+    before = function(self, dao_factory, helpers)
+      self.params.username_or_id = ngx.unescape_uri(self.params.username_or_id)
+      crud.find_consumer_by_username_or_id(self, dao_factory, helpers)
+      crud.find_plugin_by_filter(self, dao_factory, {
+        consumer_id = self.consumer.id,
+        id          = self.params.id
```

Comment:
style: such tables could use a trailing comma, as documented in our style guide, to reduce the diff size of future changes applied here.